### ISSUE TITLE
Quickly stepping through "cells" or "run selection" causes Spyder ipython hang

### ISSUE THREAD

**Issue by snowflakeas** [spyder-ide/spyder#16413]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

Sypder 5.0.3
Win10
But has existed since Spyder 4.0
Multiple PC installations

### What steps reproduce the problem?

1. Run selection / current line F9 quickly 
2. Console hangs
3. Not repeatable, but is a daily occurence.

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 5.0.3 (since 4.2 at least)
* Python version: 3.7.11 (but effects for the past year)
* Qt version: 
* PyQt version: 
* Operating System name/version: Win10

### Dependencies

```
PASTE DEPENDENCIES HERE
```

**Comment by steff456** [comment]
Hi @snowflakeas,

Can you please upload a gif with the behavior you are having? How many times are you pressing F9 before the console starts hanging?

**Comment by snowflakeas** [comment]
Hi - 
Hard to make a gif . as it is not repeatable - occurs both with "exec current line" and "exec. current cell".
Made a quick test:

```
# -*- coding: utf-8 -*-
"""
Created on Tue Sep 14 07:52:11 2021

@author: admin
"""
from scipy import signal
import scipy.fftpack as fft
import numpy as np
from numpy import pi, log10, ceil, sqrt, exp
from numpy import pi, sin, cos, rad2deg, deg2rad, log10
import matplotlib.pyplot as plt
import sys
import os

# %%

plt.close("all")

fs = 500e6
t_dwell = 200e-9
t = np.linspace(0, t_dwell, int(t_dwell*fs)+1)
f = 100e6
y = sin(2*pi*f*t)
f_sweep = np.linspace(30, 140, int((140-30)/10)+1)*1e6
t_delay = 5e-9  # ns
delay = exp(-1j*2*pi*f_sweep/fs*fs*t_delay)

# plt.plot(t, y)
plt.plot(f_sweep/1e6, np.arctan2(delay.imag, delay.real)*180/pi,
         label="phase added by %.0f ns delay" % (t_delay*1e9))
plt.ylabel("phase (deg)")
plt.xlabel("freq (MHz)")
plt.legend()

plt.figure()
# plt.plot(t,y)
# plt.plot(t,y*(exp(-1j*2*pi*f/fs*fs*t_delay)).imag)
yy = exp(1j*2*pi*f*t)
yy_delay = (yy * exp(-1j*2*pi*f/fs*fs*t_delay))
# plt.plot(t,y*(exp(-1j*2*pi*f/fs*fs*t_delay)).imag)
plt.plot(t*1e9, yy.imag, label="%.1fMHz sin" % (f/1e6))
plt.plot(t*1e9, yy_delay.imag, label="with %.0f ns delay" % (t_delay * 1e9))
plt.ylabel("amplitude")
plt.xlabel("time (ns)")
plt.legend()

```

Ran this in ipython session to the right 
runcell(0,...) 
runcell(1,...) 
clicking on the "run cell" button" (or via Ctrl-Enter)

This morning after running cell 1 the 28th time the ipython command prompt stopped responding (waited 1 minute)
Had to restart kernel.

But this is not easily repeatable (sometimes it works for hundreds of runs) - but some of my data comes from measurements being taking in the script - single when single steps fail then I loose my interactive measurement data.

**Comment by ccordoba12** [comment]
Did you change any option related to running code in Spyder? Or are you using its factory defaults?

**Comment by snowflakeas** [comment]
Sorry for the delay in response. After 2 years of Spyder I have moved over to VS Code. 

Pretty much use factory defaults - except that plots windows are external (not inline, not in tabs)

Spyder 4 & 5 were slow on all my Win10 machine installations (on modern CPUs, with SSD) , unless I turned off all default linting etc. after an hour of work even typing in the editor would be subject to 300-600ms lag between keystrokes.
(Colleagues in completely different fields / completely different setups have also commented how laggy Spyder feels)

The intermittent hanging when running cells in rapid succession was also a factor in moving away from Spyder 

Wish you all the best with the project.
